control-plane/rest/openapi-specs/v0_api_spec.yaml

```diff
@@ -4344,6 +4355,81 @@ components:
           type: integer
       required:
         - entries
+    Encryption:
+      description: Encryption secret details or the encryption params.
+      type: object
+      properties:
+        secret:
+          $ref: '#/components/schemas/EncryptionSecret'
+        data:
+          $ref: '#/components/schemas/EncryptionData'
+      oneOf:
+        - required:
+            - secret
+        - required:
+            - data
+    EncryptionSecret:
+      description: Encryption secret details.
+      type: object
+      properties:
+        secret:
+          description: Encryption secret name, could be a file name.
+          type: string
+          example: pool-secret-encr
+      required:
+        - secret
+    EncryptionData:
+      description: Encryption parameters.
+      type: object
+      properties:
+        cipher:
+          description: Cipher to use for encryption.
+          example: AES_XTS
+          type: string
+          enum:
+            - AES_CBC
+            - AES_XTS
+        key:
+          $ref: '#/components/schemas/EncryptionKey'
+      required:
+        - cipher
+        - key
+    EncryptionKey:
+      description: Represents an encryption key that can be used to encrypt an entity like pool or replica.
+      type: object
+      properties:
+        key_name:
+          description: Name of the key.
+          example: encrkey
+          type: string
+        key:
+          description: The AES encryption key.
+          example: [ 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 49, 50, 51, 52, 53, 54, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 100, 101, 97, 100, 99, 48 ]
+          type: array
```

Comment:
Should this be a string?
How do expect users to generate the secret key? Do we also ask them to md5sum it?

Reply:
The way for users to supply it is via a secret or a file, and both of them have string in it. The caller of the secret provider crate would have to handle that.

Reply:
Even for testing, isn't a string easier?

Reply:
Sure. Will make it a string.